assign-goals.lua

```diff
@@ -99,9 +99,10 @@ function assign(goals, unit, reset)
         assert(type(realized) == "boolean")
         goal = goal:upper()
         if df.goal_type[goal] then
-            utils.insert_or_update(unit.status.current_soul.personality.dreams,
-                                   { new = true, type = df.goal_type[goal], unk8 = realized and 1 or 0 },
+            local _, _, pos = utils.insert_or_update(unit.status.current_soul.personality.dreams,
+                                   { new = true, type = df.goal_type[goal] },
```

Comment:
Recursive assignment is supported here, so I would suggest adding `flags = {accomplished = realized}` to the table here instead.

Reply:
Done!